File: problems/programmers/42584-stock-price/younho9.js
function solution(prices) {
    const answer = Array(prices.length);
    const s = new Stack();

    prices.forEach((price, i) => {
        while (!s.isEmpty() && prices[s.top()] > price) {
            answer[s.top()] = i - s.top();
            s.pop();
        }
        s.push(i);
    });

    while (!s.isEmpty()) {
        answer[s.top()] = prices.length - s.top() - 1;
        s.pop();
    }

    return answer;
}

class Stack {
    constructor() {
        this._arr = [];
    }

    push(value) {
        this._arr.push(value);
    }

    pop() {
        return this._arr.pop();
    }

    top() {
        return this._arr[this._arr.length - 1];
    }

    isEmpty() {
        return this._arr.length === 0;
    }
}

module.exports = solution;

File: problems/programmers/42584-stock-price/younho9.py
def solution(prices):
    answer = [0] * len(prices)
    stack = []

    for i in range(len(prices)):
        while stack and prices[stack[len(stack) - 1]] > prices[i]:
            answer[stack[len(stack) - 1]] = i - stack[len(stack) - 1]
            stack.pop()
        stack.append(i)

    while stack:
        answer[stack[len(stack) - 1]] = len(prices) - stack[len(stack) - 1] - 1
        stack.pop()

    return answer
